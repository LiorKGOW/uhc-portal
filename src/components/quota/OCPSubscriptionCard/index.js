import OCPSubscriptionCard from './OCPSubscriptionCard';

export default OCPSubscriptionCard;
