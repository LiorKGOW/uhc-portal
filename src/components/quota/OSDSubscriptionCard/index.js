import { connect } from 'react-redux';

import { subscriptionsActions } from '../../../redux/actions/subscriptionsActions';
import OSDSubscriptionCard from './OSDSubscriptionCard';

import { MARKETPLACE_QUOTA_FEATURE } from '../../../redux/constants/featureConstants';

const mapDispatchToProps = {
  fetchQuotaCost: organizationID => subscriptionsActions.fetchQuotaCost(organizationID),
};

function mapStateToProps(state) {
  return {
    quotaCost: state.subscriptions.quotaCost,
    marketplaceQuotaFeature: state.features[MARKETPLACE_QUOTA_FEATURE],
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(OSDSubscriptionCard);
