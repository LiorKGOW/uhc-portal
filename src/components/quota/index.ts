import { connect } from 'react-redux';
import { subscriptionsActions } from '../../redux/actions/subscriptionsActions';
import { clustersActions } from '../../redux/actions/clustersActions';
import Quota from './Quota';
import { GlobalState } from '~/redux/store';

const mapDispatchToProps = {
  fetchAccount: () => subscriptionsActions.fetchAccount(),
  invalidateClusters: () => clustersActions.invalidateClusters(),
};

const mapStateToProps = (state: GlobalState) => ({
  account: state.subscriptions.account,
});

export default connect(mapStateToProps, mapDispatchToProps)(Quota);
