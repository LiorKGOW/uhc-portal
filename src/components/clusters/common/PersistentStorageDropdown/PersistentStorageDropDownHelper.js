
import get from 'lodash/get';
import has from 'lodash/has';
import { parseValueWithUnit } from '../../../../common/units';

import { billingModels } from '../../../../common/subscriptionTypes';

/**
 * Returns last level indexed by resource_name e.g. {'gp2': 2700}.
 */
const quotaLookup = (cluster, quota) => {
  const billingModel = get(cluster, 'billing_model', billingModels.STANDARD);
  const product = cluster.subscription.plan.id; // TODO plan.type
  const cloudProviderID = get(cluster, 'cloud_provider.id', 'any');
  const infra = cluster.ccs.enabled ? 'byoc' : 'rhInfra';
  const zoneType = cluster.multi_az ? 'multiAZ' : 'singleAZ';
  const resourceName = 'gp2';
  return get(quota.storageQuota,
    [billingModel, product, cloudProviderID, infra, zoneType, resourceName], {});
};


// Storage quota is only visible if it has an entry in the quota summary
// regardless of whether the org has quota or not
const isAvailable = (cluster, organization, quota) => {
  // We get quota together with organization.
  // TODO: have action/reducer set quota.fullfilled, drop organization arg.
  if (!organization.fulfilled) {
    return false;
  }

  // If the storage quota is not in the quota cost, it should not be available
  return has(quotaLookup(cluster, quota));
};

const baseClusterQuota = 107374182400; // The base cluster storage quota is 100 GiB (in bytes).

const filterPersistentStorageValuesByQuota = (currentValue,
  persistentStorageValues, remainingQuota) => {
  // Get quota for persistent storage.
  // this quota is "on top" of the base cluster quota of 100 GiB.
  const quotaInBytes = parseValueWithUnit(remainingQuota, 'GiB');
  const result = { ...persistentStorageValues };
  if (!result.values) {
    return { values: [{ value: baseClusterQuota, unit: 'B' }] };
  }
  if (currentValue) {
    result.values = result.values.filter(el => el.value <= quotaInBytes + currentValue);
  } else {
    result.values = result.values.filter(el => el.value <= quotaInBytes + baseClusterQuota);
  }
  return result;
};


export {
  quotaLookup,
  isAvailable,
  filterPersistentStorageValuesByQuota,
};
