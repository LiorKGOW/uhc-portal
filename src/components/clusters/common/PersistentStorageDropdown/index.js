import get from 'lodash/get';
import { connect } from 'react-redux';

import getPersistentStorage from '../../../../redux/actions/persistentStorageActions';
import PersistentStorageDropdown from './PersistentStorageDropdown';

const mapStateToProps = state => ({
  persistentStorageValues: state.persistentStorageValues,
  quota: get(state, 'userProfile.organization.quotaList.storageQuota.aws.available', 0),
});

const mapDispatchToProps = { getPersistentStorage };

export default connect(mapStateToProps, mapDispatchToProps)(PersistentStorageDropdown);
