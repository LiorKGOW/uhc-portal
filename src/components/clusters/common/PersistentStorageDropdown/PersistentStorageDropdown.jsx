// PersistentStorageDropdown shows a selection of storage options
// for setting on the installed a cluster.
// it is meant to be used in a redux-form <Field> and expects an onChange callback.

import React from 'react';
import PropTypes from 'prop-types';
import {
  FormSelect,
  FormSelectOption,
  Tooltip,
} from '@patternfly/react-core';

import { Spinner } from '@redhat-cloud-services/frontend-components';
import ErrorBox from '../../../common/ErrorBox';
import { humanizeValueWithUnitGiB } from '../../../../common/units';
import filterPersistentStorageValuesByQuota from './helpers';
import { noQuotaTooltip } from '../../../../common/helpers';

class PersistentStorageDropdown extends React.Component {
  componentDidMount() {
    const {
      getPersistentStorage, persistentStorageValues,
    } = this.props;
    if (!persistentStorageValues.pending
      && !persistentStorageValues.fulfilled
      && !persistentStorageValues.error) {
      // fetch persistent storage from server only if needed.
      getPersistentStorage();
    }
  }

  render() {
    const {
      input, persistentStorageValues, disabled, currentValue, quota,
    } = this.props;

    // Set up options for storage values
    const storageOption = (value) => {
      // value is a tuple of {value, unit}
      const valueWithUnit = humanizeValueWithUnitGiB(value.value);
      const strValue = value.value.toString();
      // Values passed in the select are *always* in bytes, but we display them
      // in a humanize form.

      return (
        <FormSelectOption
          key={strValue}
          value={strValue}
          label={`${valueWithUnit.value} ${valueWithUnit.unit}`}
        />
      );
    };

    if (persistentStorageValues.fulfilled) {
      const filteredStorageValues = filterPersistentStorageValuesByQuota(currentValue,
        persistentStorageValues, quota);
      const notEnoughQuota = filteredStorageValues.values.length <= 1;
      const isDisabled = disabled || notEnoughQuota;
      const formSelect = (
        <FormSelect
          className="quota-dropdown"
          aria-label="Persistent Storage"
          isDisabled={isDisabled}
          {...input}
        >
          {filteredStorageValues.values.map(value => storageOption(value))}
        </FormSelect>
      );

      if (notEnoughQuota) {
        return (
          <Tooltip
            content={noQuotaTooltip}
            position="right"
          >
            <div>
              {formSelect}
            </div>
          </Tooltip>
        );
      }
      return formSelect;
    }

    return persistentStorageValues.error ? (
      <ErrorBox message="Error loading persistent storage list" response={persistentStorageValues} />
    ) : (
      <>
        <div className="spinner-fit-container"><Spinner /></div>
        <div className="spinner-loading-text">Loading persistent storage list...</div>
      </>
    );
  }
}

PersistentStorageDropdown.propTypes = {
  getPersistentStorage: PropTypes.func.isRequired,
  persistentStorageValues: PropTypes.object.isRequired,
  input: PropTypes.object.isRequired,
  disabled: PropTypes.bool.isRequired,
  quota: PropTypes.number.isRequired,
  currentValue: PropTypes.number,
};

export default PersistentStorageDropdown;
