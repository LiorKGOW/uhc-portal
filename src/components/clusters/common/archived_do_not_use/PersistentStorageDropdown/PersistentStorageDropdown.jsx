// PersistentStorageDropdown shows a selection of storage options
// for setting on the installed a cluster.
// it is meant to be used in a redux-form <Field> and expects an onChange callback.

import React from 'react';
import PropTypes from 'prop-types';

import { FormSelect, FormSelectOption, Tooltip } from '@patternfly/react-core';
import { Spinner } from '@redhat-cloud-services/frontend-components/Spinner';

import { noQuotaTooltip } from '../../../../../common/helpers';
import { humanizeValueWithUnitGiB } from '../../../../../common/units';
import ErrorBox from '../../../../common/ErrorBox';
import { QuotaTypes } from '../../quotaModel';
import { availableQuota } from '../../quotaSelectors';

import { filterPersistentStorageValuesByQuota } from './PersistentStorageDropDownHelper';

class PersistentStorageDropdown extends React.Component {
  componentDidMount() {
    const { getPersistentStorage, persistentStorageValues } = this.props;
    if (
      !persistentStorageValues.pending &&
      !persistentStorageValues.fulfilled &&
      !persistentStorageValues.error
    ) {
      // fetch persistent storage from server only if needed.
      getPersistentStorage();
    }
  }

  render() {
    const {
      input,
      persistentStorageValues,
      disabled,
      currentValue,
      quotaList,
      billingModel,
      product,
      cloudProviderID,
      isBYOC,
      isMultiAZ,
    } = this.props;

    // Set up options for storage values
    const storageOption = (value) => {
      // value is a tuple of {value, unit}
      const valueWithUnit = humanizeValueWithUnitGiB(value.value);
      const strValue = value.value.toString();
      // Values passed in the select are *always* in bytes, but we display them
      // in a humanize form.

      return (
        <FormSelectOption
          key={strValue}
          value={strValue}
          label={`${valueWithUnit.value} ${valueWithUnit.unit}`}
        />
      );
    };

    if (persistentStorageValues.fulfilled) {
      const storageQuota = availableQuota(quotaList, {
        resourceType: QuotaTypes.STORAGE,
        billingModel,
        product,
        cloudProviderID,
        isBYOC,
        isMultiAZ,
        resourceName: 'gp2',
      });

      const filteredStorageValues = filterPersistentStorageValuesByQuota(
        currentValue,
        persistentStorageValues,
        storageQuota,
      );
      const notEnoughQuota = filteredStorageValues.values.length <= 1;
      const isDisabled = disabled || notEnoughQuota;

      const { onChange, ...restInput } = input;
      const formSelect = (
        <FormSelect
          className="quota-dropdown"
          aria-label="Persistent Storage"
          isDisabled={isDisabled}
          onChange={(_event, value) => onChange(value)}
          {...restInput}
        >
          {filteredStorageValues.values.sort((a, b) => a.value - b.value).map(storageOption)}
        </FormSelect>
      );

      if (notEnoughQuota) {
        return (
          <Tooltip content={noQuotaTooltip} position="right">
            <div>{formSelect}</div>
          </Tooltip>
        );
      }
      return formSelect;
    }

    return persistentStorageValues.error ? (
      <ErrorBox
        message="Error loading persistent storage list"
        response={persistentStorageValues}
      />
    ) : (
      <>
        <div className="spinner-fit-container">
          <Spinner />
        </div>
        <div className="spinner-loading-text">Loading persistent storage list...</div>
      </>
    );
  }
}

PersistentStorageDropdown.propTypes = {
  getPersistentStorage: PropTypes.func.isRequired,
  persistentStorageValues: PropTypes.object.isRequired,
  input: PropTypes.object.isRequired,
  disabled: PropTypes.bool,
  quotaList: PropTypes.object.isRequired,
  currentValue: PropTypes.number,
  billingModel: PropTypes.string.isRequired,
  product: PropTypes.string.isRequired,
  cloudProviderID: PropTypes.string.isRequired,
  isBYOC: PropTypes.bool.isRequired,
  isMultiAZ: PropTypes.bool.isRequired,
};

export default PersistentStorageDropdown;
