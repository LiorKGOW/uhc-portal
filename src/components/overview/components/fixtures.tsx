import React from 'react';

import SomeLogo from '~/styles/images/RedHatOpenShiftGitOpsLogo.svg';

import { DRAWER_PANEL_CONTENT, DrawerPanelContentNode } from './common/DrawerPanelContent';
import PRODUCT_CARD_LOGOS from './common/ProductCardLogos';
import { FEATURED_PRODUCTS_CARDS } from './FeaturedProductsCards/FeaturedProductsCards';

const BASIC_PRODUCT_CARD_PROPS = {
  title: 'example title',
  description: 'example description',
  logo: 'product-overview-card__logo',
  labelText: 'example labelText',
  drawerPanelContent: {
    head: <div data-testid="test-head-div">example drawer panel content head</div>,
    body: <div data-testid="test-body-div">example drawer panel content body</div>,
  },
  openLearnMore: jest.fn(),
  isSelected: false,
  dataTestId: 'some-id',
};

const PRODUCT_CARD_TEST_CASES = {
  UNSELECTED: {
    ...BASIC_PRODUCT_CARD_PROPS,
  },

  SELECTED: {
    ...BASIC_PRODUCT_CARD_PROPS,
    isSelected: true,
  },

  WITHOUT_LABEL_TEXT: {
    ...BASIC_PRODUCT_CARD_PROPS,
    labelText: undefined,
  },
};

const BASIC_DRAWER_PANEL_PROPS = {
  children: <div data-testid="children-of-drawer-panel">some children</div>,
  content: {
    head: <div data-testid="test-head-div">example drawer panel content head</div>,
    body: <div data-testid="test-body-div">example drawer panel content body</div>,
  },
  isOpen: true,
  onClose: jest.fn(),
};

const DRAWER_PANEL_TEST_CASES = {
  OPENED: { ...BASIC_DRAWER_PANEL_PROPS },
};

const BASIC_RECCOMENDED_OPERATORS_CARDS_PROPS = {
  openLearnMore: jest.fn(),
  selectedCardTitle: '',
};

const RECOMMENDED_OPERATORS_CARDS_TEST_CASES = {
  NON_SELECTED: {
    ...BASIC_RECCOMENDED_OPERATORS_CARDS_PROPS,
  },
  GITOPS_SELECTED: {
    ...BASIC_RECCOMENDED_OPERATORS_CARDS_PROPS,
    selectedCardTitle: PRODUCT_CARD_LOGOS.gitops.title,
  },
};

type RecommendedOperatorsCardsNode = {
  title: string;
  description: string;
  logo: string | undefined;
  labelText?: string;
  drawerPanelContent?: DrawerPanelContentNode;
};

const RECOMMENDED_OPERATORS_CARDS_DATA: RecommendedOperatorsCardsNode[] = [
  {
    ...PRODUCT_CARD_LOGOS.gitops,
    description:
      'Integrate git repositories, continuous integration/continuous delivery (CI/CD) tools, and Kubernetes.',
    labelText: 'Free',
    drawerPanelContent: DRAWER_PANEL_CONTENT.gitops,
  },
  {
    ...PRODUCT_CARD_LOGOS.pipelines,
    description:
      'Automate your application delivery using a continuous integration and continuous deployment (CI/CD) framework.',
    labelText: 'Free',
    drawerPanelContent: DRAWER_PANEL_CONTENT.pipelines,
  },
  {
    ...PRODUCT_CARD_LOGOS.serviceMesh,
    description: 'Connect, manage, and observe microservices-based applications in a uniform way.',
    labelText: 'Free',
    drawerPanelContent: DRAWER_PANEL_CONTENT.serviceMesh,
  },
];

const BASIC_FEATURED_PRODUCTS_CARDS_PROPS = {
  openLearnMore: jest.fn(),
  selectedCardTitle: '',
};

const FEATURED_PRODUCTS_CARDS_TEST_CASES = {
  NON_SELECTED: {
    ...BASIC_FEATURED_PRODUCTS_CARDS_PROPS,
  },
  ADVANCED_CLUSTER_SECURITY_SELECTED: {
    ...BASIC_FEATURED_PRODUCTS_CARDS_PROPS,
    selectedCardTitle: PRODUCT_CARD_LOGOS.advancedClusterSecurity.title,
  },
};

const GENERAL_DRAWER_PANEL_HEAD_BASIC = {
  title: 'some title',
  logo: SomeLogo,
};

const GENERAL_DRAWER_PANEL_HEAD_WITH_LINK = {
  ...GENERAL_DRAWER_PANEL_HEAD_BASIC,
  trialButtonLink: 'https://www.redhat.com/en',
};

export {
  PRODUCT_CARD_TEST_CASES,
  DRAWER_PANEL_TEST_CASES,
  RECOMMENDED_OPERATORS_CARDS_TEST_CASES,
  RECOMMENDED_OPERATORS_CARDS_DATA,
  FEATURED_PRODUCTS_CARDS_TEST_CASES,
  FEATURED_PRODUCTS_CARDS,
  GENERAL_DRAWER_PANEL_HEAD_BASIC,
  GENERAL_DRAWER_PANEL_HEAD_WITH_LINK,
};
