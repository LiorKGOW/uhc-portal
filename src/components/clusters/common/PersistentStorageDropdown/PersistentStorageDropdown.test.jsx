import React from 'react';
import { mount } from 'enzyme';

import PersistentStorageDropdown from './PersistentStorageDropdown';
import fixtures from '../../ClusterDetails/__test__/ClusterDetails.fixtures';
import { storageQuotaList } from '../__test__/quota.fixtures';

const baseState = {
  error: false,
  errorMessage: '',
  pending: false,
  fulfilled: false,
};

describe('<PersistentStorageDropdown />', () => {
  describe('when persistent storage list needs to be fetched', () => {
    let getPersistentStorage;
    let onChange;
    let wrapper;
    beforeEach(() => {
      getPersistentStorage = jest.fn();
      onChange = jest.fn();
      wrapper = mount(
        <PersistentStorageDropdown
          persistentStorageValues={baseState}
          input={{ onChange }}
          getPersistentStorage={getPersistentStorage}
          quotaList={storageQuotaList}
          product={fixtures.clusterDetails.cluster.subscription.plan.type}
          cloudProviderID={fixtures.clusterDetails.cluster.cloud_provider.id}
          billingModel="standard"
          isBYOC={fixtures.clusterDetails.cluster.ccs.enabled}
          isMultiAZ={fixtures.clusterDetails.cluster.multi_az}
          disabled={false}
        />,
      );
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('calls getPersistentStorage', () => {
      expect(getPersistentStorage).toBeCalled();
    });
  });

  describe('when there was an error', () => {
    let getPersistentStorage;
    let onChange;
    let wrapper;
    beforeEach(() => {
      const state = {
        ...baseState,
        error: true,
        errorMessage: 'This is an error message',
      };

      getPersistentStorage = jest.fn();
      onChange = jest.fn();
      wrapper = mount(
        <PersistentStorageDropdown
          persistentStorageValues={state}
          input={{ onChange }}
          getPersistentStorage={getPersistentStorage}
          quotaList={storageQuotaList}
          product={fixtures.clusterDetails.cluster.subscription.plan.type}
          cloudProviderID={fixtures.clusterDetails.cluster.cloud_provider.id}
          billingModel="standard"
          isBYOC={fixtures.clusterDetails.cluster.ccs.enabled}
          isMultiAZ={fixtures.clusterDetails.cluster.multi_az}
          disabled={false}
        />,
      );
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when the request is pending', () => {
    let getPersistentStorage;
    let onChange;
    let wrapper;
    const state = {
      error: false,
      errorMessage: '',
      pending: true,
      fulfilled: false,
      values: [],
    };
    beforeEach(() => {
      getPersistentStorage = jest.fn();
      onChange = jest.fn();
      wrapper = mount(
        <PersistentStorageDropdown
          persistentStorageValues={state}
          input={{ onChange }}
          getPersistentStorage={getPersistentStorage}
          quotaList={storageQuotaList}
          product={fixtures.clusterDetails.cluster.subscription.plan.type}
          cloudProviderID={fixtures.clusterDetails.cluster.cloud_provider.id}
          billingModel="standard"
          isBYOC={fixtures.clusterDetails.cluster.ccs.enabled}
          isMultiAZ={fixtures.clusterDetails.cluster.multi_az}
          disabled={false}
        />,
      );
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('does not call getPersistentStorage again if request returns an error', () => {
      wrapper.setProps({
        persistentStorageValues: { ...state, error: true, pending: false },
      }, () => {
        expect(getPersistentStorage).not.toBeCalled();
      });
    });
  });

  describe('when the storage list is available', () => {
    let getPersistentStorage;
    let onChange;
    let wrapper;
    beforeEach(() => {
      const state = {
        ...baseState,
        fulfilled: true,
        values: [{ unit: 'B', value: 107374182400 }, { unit: 'B', value: 644245094400 }, { unit: 'B', value: 1181116006400 }],
      };

      getPersistentStorage = jest.fn();
      onChange = jest.fn();
      wrapper = mount(
        <PersistentStorageDropdown
          persistentStorageValues={state}
          input={{ onChange }}
          getPersistentStorage={getPersistentStorage}
          quotaList={storageQuotaList}
          product={fixtures.clusterDetails.cluster.subscription.plan.type}
          cloudProviderID={fixtures.clusterDetails.cluster.cloud_provider.id}
          billingModel="standard"
          isBYOC={fixtures.clusterDetails.cluster.ccs.enabled}
          isMultiAZ={fixtures.clusterDetails.cluster.multi_az}
          disabled={false}
        />,
      );
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
