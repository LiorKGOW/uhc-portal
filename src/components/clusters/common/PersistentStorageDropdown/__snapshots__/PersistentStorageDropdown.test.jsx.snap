// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersistentStorageDropdown /> when persistent storage list needs to be fetched renders correctly 1`] = `
<PersistentStorageDropdown
  disabled={false}
  getPersistentStorage={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  input={
    Object {
      "onChange": [MockFunction],
    }
  }
  organization={
    Object {
      "fulfilled": true,
      "pending": false,
    }
  }
  persistentStorageValues={
    Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "values": Array [],
    }
  }
  quota={Object {}}
  storageQuota={0}
>
  <div
    className="spinner-fit-container"
  >
    <Spinner>
      <div
        className="ins-c-spinner"
        role="status"
      >
        <span
          className="pf-u-screen-reader"
        >
          Loading...
        </span>
      </div>
    </Spinner>
  </div>
  <div
    className="spinner-loading-text"
  >
    Loading persistent storage list...
  </div>
</PersistentStorageDropdown>
`;

exports[`<PersistentStorageDropdown /> when the request is pending renders correctly 1`] = `
<PersistentStorageDropdown
  disabled={false}
  getPersistentStorage={[MockFunction]}
  input={
    Object {
      "onChange": [MockFunction],
    }
  }
  organization={
    Object {
      "fulfilled": true,
      "pending": false,
    }
  }
  persistentStorageValues={
    Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": true,
      "values": Array [],
    }
  }
  storageQuota={0}
>
  <div
    className="spinner-fit-container"
  >
    <Spinner>
      <div
        className="ins-c-spinner"
        role="status"
      >
        <span
          className="pf-u-screen-reader"
        >
          Loading...
        </span>
      </div>
    </Spinner>
  </div>
  <div
    className="spinner-loading-text"
  >
    Loading persistent storage list...
  </div>
</PersistentStorageDropdown>
`;

exports[`<PersistentStorageDropdown /> when the storage list is available renders correctly 1`] = `
<PersistentStorageDropdown
  disabled={false}
  getPersistentStorage={[MockFunction]}
  input={
    Object {
      "onChange": [MockFunction],
    }
  }
  organization={
    Object {
      "fulfilled": true,
      "pending": false,
    }
  }
  persistentStorageValues={
    Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": true,
      "pending": false,
      "values": Array [
        Object {
          "unit": "B",
          "value": 107374182400,
        },
        Object {
          "unit": "B",
          "value": 644245094400,
        },
        Object {
          "unit": "B",
          "value": 1181116006400,
        },
      ],
    }
  }
  storageQuota={600}
>
  <FormSelect
    aria-label="Persistent Storage"
    className="quota-dropdown"
    isDisabled={false}
    isRequired={false}
    onBlur={[Function]}
    onChange={[MockFunction]}
    onFocus={[Function]}
    ouiaSafe={true}
    validated="default"
    value=""
  >
    <select
      aria-invalid={false}
      aria-label="Persistent Storage"
      className="pf-c-form-control quota-dropdown"
      data-ouia-component-id="OUIA-Generated-FormSelect-default-1"
      data-ouia-component-type="PF4/FormSelect"
      data-ouia-safe={true}
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      value=""
    >
      <FormSelectOption
        key="107374182400"
        label="100 GiB"
        value="107374182400"
      >
        <option
          className=""
          disabled={false}
          value="107374182400"
        >
          100 GiB
        </option>
      </FormSelectOption>
      <FormSelectOption
        key="644245094400"
        label="600 GiB"
        value="644245094400"
      >
        <option
          className=""
          disabled={false}
          value="644245094400"
        >
          600 GiB
        </option>
      </FormSelectOption>
    </select>
  </FormSelect>
</PersistentStorageDropdown>
`;

exports[`<PersistentStorageDropdown /> when there was an error renders correctly 1`] = `
<PersistentStorageDropdown
  disabled={false}
  getPersistentStorage={[MockFunction]}
  input={
    Object {
      "onChange": [MockFunction],
    }
  }
  organization={
    Object {
      "fulfilled": true,
      "pending": false,
    }
  }
  persistentStorageValues={
    Object {
      "error": true,
      "errorMessage": "This is an error message",
      "fulfilled": false,
      "pending": false,
      "values": Array [],
    }
  }
  storageQuota={0}
>
  <ErrorBox
    message="Error loading persistent storage list"
    response={
      Object {
        "error": true,
        "errorMessage": "This is an error message",
        "fulfilled": false,
        "pending": false,
        "values": Array [],
      }
    }
  >
    <Alert
      className="error-box"
      isInline={true}
      title="Error loading persistent storage list"
      variant="danger"
    >
      <div
        aria-label="Danger Alert"
        className="pf-c-alert pf-m-inline pf-m-danger error-box"
        data-ouia-component-id="OUIA-Generated-Alert-danger-1"
        data-ouia-component-type="PF4/Alert"
        data-ouia-safe={true}
      >
        <Component
          variant="danger"
        >
          <div
            className="pf-c-alert__icon"
          >
            <ExclamationCircleIcon
              color="currentColor"
              noVerticalAlign={false}
              size="sm"
            >
              <svg
                aria-hidden={true}
                aria-labelledby={null}
                fill="currentColor"
                height="1em"
                role="img"
                style={
                  Object {
                    "verticalAlign": "-0.125em",
                  }
                }
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                />
              </svg>
            </ExclamationCircleIcon>
          </div>
        </Component>
        <h4
          className="pf-c-alert__title"
        >
          <span
            className="pf-u-screen-reader"
          >
            Danger alert:
          </span>
          Error loading persistent storage list
        </h4>
        <div
          className="pf-c-alert__description"
        >
          <span>
            This is an error message
          </span>
          <br />
          <span>
            Operation ID: N/A
          </span>
        </div>
      </div>
    </Alert>
  </ErrorBox>
</PersistentStorageDropdown>
`;
